src/App.js

```diff
@@ -35,7 +35,7 @@ import MyToolbar from "./components/Toolbar";
 import { withStyles } from "@material-ui/core/styles";
 import AppBar from "@material-ui/core/AppBar";
 
-
+var delayTimer;
```

Comment:
I'm not sure about js global variables scoping, but maybe it is better to rename this to something like "searchDelayTimer"
```suggestion
var searchDelayTimer;
```

Can you also move it just before `class App` please?